Source/Logger/trace.cpp

```diff
@@ -221,6 +319,8 @@ void TraceState::Init() noexcept
     if (previousCount == 0)
     {
         m_initTime = std::chrono::high_resolution_clock::now();
+
+        RegisterProvider();
```

Comment:
>RegisterProvider(); [](http://example.com/codeflow?start=8&length=19)

Curious about the best time to call this.  I think ideally we wouldn't register the provider if the client hasn't registered ETW.  Maybe move to HCTraceSetEtwEnabled?

Reply:
Ah that's a better idea, yeah